tests/e2e/nvidia-ctk_containerd_test.go

```diff
@@ -0,0 +1,415 @@
+
+		// Backup original containerd configuration
+		_, _, err = nestedContainerRunner.Run(`
+			# Backup the original conf.d directory
+			if [ -d /etc/containerd/conf.d ]; then
+				cp -r /etc/containerd/conf.d /tmp/containerd-conf.d.backup
+			fi
+		`)
+		Expect(err).ToNot(HaveOccurred(), "Failed to backup containerd configuration")
+
+		// Install the NVIDIA Container Toolkit packages
+		_, _, err = toolkitInstaller.Install(nestedContainerRunner)
+		Expect(err).ToNot(HaveOccurred(), "Failed to install toolkit for containerd")
+	})
+
+	AfterAll(func(ctx context.Context) {
+		// Cleanup: remove the container
+		_, _, err := runner.Run(fmt.Sprintf("docker rm -f %s 2>/dev/null || true", containerName))
+		if err != nil {
+			GinkgoLogr.Error(err, "failed to cleanup container", "container", containerName)
+		}
+	})
+
+	BeforeEach(func(ctx context.Context) {
+		// No setup needed - each test starts with the state from the previous test
+	})
+
+	AfterEach(func(ctx context.Context) {
+		// Step 1: Restore containerd configuration from backup
+		_, _, err := nestedContainerRunner.Run(`
+			# Restore the original conf.d
+			if [ -d /tmp/containerd-conf.d.backup ]; then
+				rm -rf /etc/containerd/conf.d
+				cp -r /tmp/containerd-conf.d.backup /etc/containerd/conf.d
+			else
+				# If no backup exists, just clean up
+				rm -rf /etc/containerd/conf.d
+				mkdir -p /etc/containerd/conf.d
+			fi
+		`)
+		Expect(err).ToNot(HaveOccurred(), "Failed to restore containerd configuration")
+
+		// Step 2: Restart containerd and wait for it to be ready
+		err = restartContainerdAndWait(nestedContainerRunner)
+		Expect(err).ToNot(HaveOccurred(), "Failed to restart containerd after configuration restore")
+	})
+
+	When("configuring containerd on a KIND node", func() {
+		It("should add NVIDIA runtime using drop-in config without modifying the main config", func(ctx context.Context) {
+			// Configure containerd using nvidia-ctk
+			_, _, err := nestedContainerRunner.Run(`nvidia-ctk runtime configure --runtime=containerd --config=/etc/containerd/config.toml --drop-in-config=/etc/containerd/conf.d/99-nvidia.toml --set-as-default --cdi.enabled`)
+			Expect(err).ToNot(HaveOccurred(), "Failed to configure containerd")
+
+			// restart containerd
+			err = restartContainerdAndWait(nestedContainerRunner)
+			Expect(err).ToNot(HaveOccurred(), "Failed to restart containerd")
+
+			// Verify containerd loaded the config correctly
+			output, _, err := nestedContainerRunner.Run(`containerd config dump`)
+			Expect(err).ToNot(HaveOccurred())
+
+			// Basic check that nvidia runtime exists
+			Expect(output).To(SatisfyAny(
+				ContainSubstring(`[plugins."io.containerd.grpc.v1.cri".containerd.runtimes.nvidia]`),
+				ContainSubstring(`[plugins.'io.containerd.cri.v1.runtime'.containerd.runtimes.nvidia]`),
+			))
+
+			// Verify the drop-in config was processed (config dump shows actual imported files)
+			Expect(output).To(ContainSubstring(`/etc/containerd/conf.d/*.toml`))
```

Comment:
Can we please load the output config as `toml` and check specific values instead of checking substrings.

Reply:
we now parse the TOML and run the checks from it